File: benchmarkDS/ZeroShot/CLAUDE_claude-sonnet-4-5-20250929/Lse6.c

#include <stdio.h>
#include <string.h>

const char* getVehicle(char vehicles[][20], int size, int index) {
    static char invalid[] = "Invalid index";
    if (index >= 0 && index < size) {
        return vehicles[index];
    }
    return invalid;
}

void printAllVehicles(char vehicles[][20], int size) {
    for (int i = 0; i < size; i++) {
        printf("%s\\n", vehicles[i]);
    }
}

int main() {
    char vehicles[][20] = {"Car", "Truck", "Motorcycle", "Bus", "Bicycle"};
    int size = 5;
    
    // Test cases
    printf("Test Case 1: Index 0\\n");
    printf("Vehicle: %s\\n", getVehicle(vehicles, size, 0));
    printf("\\n");
    
    printf("Test Case 2: Index 2\\n");
    printf("Vehicle: %s\\n", getVehicle(vehicles, size, 2));
    printf("\\n");
    
    printf("Test Case 3: Index 4\\n");
    printf("Vehicle: %s\\n", getVehicle(vehicles, size, 4));
    printf("\\n");
    
    printf("Test Case 4: Invalid index -1\\n");
    printf("Vehicle: %s\\n", getVehicle(vehicles, size, -1));
    printf("\\n");
    
    printf("Test Case 5: Invalid index 10\\n");
    printf("Vehicle: %s\\n", getVehicle(vehicles, size, 10));
    printf("\\n");
    
    printf("All vehicles:\\n");
    printAllVehicles(vehicles, size);
    
    return 0;
}

File: benchmarkDS/ZeroShot/CLAUDE_claude-sonnet-4-5-20250929/Lse6.py

def get_vehicle(vehicles, index):
    if 0 <= index < len(vehicles):
        return vehicles[index]
    return "Invalid index"

def print_all_vehicles(vehicles):
    for vehicle in vehicles:
        print(vehicle)

if __name__ == "__main__":
    vehicles = ["Car", "Truck", "Motorcycle", "Bus", "Bicycle"]
    
    # Test cases
    print("Test Case 1: Index 0")
    print("Vehicle:", get_vehicle(vehicles, 0))
    print()
    
    print("Test Case 2: Index 2")
    print("Vehicle:", get_vehicle(vehicles, 2))
    print()
    
    print("Test Case 3: Index 4")
    print("Vehicle:", get_vehicle(vehicles, 4))
    print()
    
    print("Test Case 4: Invalid index -1")
    print("Vehicle:", get_vehicle(vehicles, -1))
    print()
    
    print("Test Case 5: Invalid index 10")
    print("Vehicle:", get_vehicle(vehicles, 10))
    print()
    
    print("All vehicles:")
    print_all_vehicles(vehicles)
